support-models/src/main/scala/com/gu/salesforce/Salesforce.scala

```diff
@@ -174,13 +173,21 @@ object Salesforce {
     implicit val codec: Codec[SalesforceErrorResponse] = deriveCodec
     val expiredAuthenticationCode = "INVALID_SESSION_ID"
     val rateLimitExceeded = "REQUEST_LIMIT_EXCEEDED"
+    val readOnlyMaintenance = "INSERT_UPDATE_DELETE_NOT_ALLOWED_DURING_MAINTENANCE"
   }
 
   case class SalesforceErrorResponse(
     message: String,
     errorCode: String
   ) extends Throwable {
-    def asRetryException: RetryException = if (errorCode == expiredAuthenticationCode || errorCode == rateLimitExceeded)
+
+    val errorsToRetryUnlimited = List(
+      SalesforceErrorResponse.expiredAuthenticationCode,
+      SalesforceErrorResponse.rateLimitExceeded,
+      SalesforceErrorResponse.readOnlyMaintenance
+    )
+
+    def asRetryException: RetryException = if (errorsToRetryUnlimited.contains(errorCode))
```

Comment:
What's the downside to (temporarily over the weekend) using retry unlimited for all errors? Increased costs running lambdas repeatedly? Anything else?

Reply:
i think this is something for subs/contributions to decide (@rupertbates @tomrf1)